File: src/leetcode/PartitionString.java
package leetcode;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

class Solution {
    public List<String> partitionString(String s) {
        List<String> answers = new ArrayList<>();
        Set<String> seen = new HashSet<>();

        StringBuilder sb = new StringBuilder();
        for (char c: s.toCharArray()){
            sb.append(c);
            if (!seen.contains(sb.toString())){
                seen.add(sb.toString());
                answers.add(sb.toString());
                sb = new StringBuilder();
            }
        }
        return answers;
    }
}
}

File: src/leetcode/PartitionString.kt
package leetcode


class PartitionString {
        fun partitionString(s: String): List<String> {
            val segments: MutableList<String> = ArrayList()
            var left = 0
            var right = 1
            while (left < right && right <= s.length) {
                val str = s.substring(left, right)
                if (!segments.contains(str)) {
                    segments.add(str)
                    left = right
                    right = left + 1
                } else {
                    right += 1
                }
            }
            return segments
        }
}